stb_ds.h: compile warning "unused parameter"

Please excuse me if you may not understand or you feel bad with my poor English.

functions "stbds_probe_position" and "stbds_hmfree_func" have unused variables and the compiler issues a warning.

gcc
```
In file included from main.c:4:
stb_ds.h: In function ‘stbds_probe_position’:
stb_ds.h:756:75: warning: unused parameter ‘slot_log2’ [-Wunused-parameter]
 static size_t stbds_probe_position(size_t hash, size_t slot_count, size_t slot_log2)
                                                                    ~~~~~~~^~~~~~~~~
In file included from main.c:4:
stb_ds.h: In function ‘stbds_hmfree_func’:
stb_ds.h:1098:57: warning: unused parameter ‘keyoff’ [-Wunused-parameter]
 void stbds_hmfree_func(void *a, size_t elemsize, size_t keyoff)
                                                  ~~~~~~~^~~~~~
```

clang
```
In file included from main.c:4:
./stb_ds.h:756:75: warning: unused parameter 'slot_log2' [-Wunused-parameter]
static size_t stbds_probe_position(size_t hash, size_t slot_count, size_t slot_log2)
                                                                          ^
./stb_ds.h:1098:57: warning: unused parameter 'keyoff' [-Wunused-parameter]
void stbds_hmfree_func(void *a, size_t elemsize, size_t keyoff)
                                                        ^
```

Is there any reason not to suppress warnings using features such as gcc extension "\__attribute\__" or c ++ 17 "[[maybe_unused]]"?

example:
```
#if __cplusplus >= 201703L
  #define STBDS_MAYBE_UNUSED(x) [[maybe_unused]] x
#elif __GNUC__
  #define STBDS_MAYBE_UNUSED(x) x __attribute__((unused))
#elif __clang__
  #define STBDS_MAYBE_UNUSED(x) x __attribute__((unused))
#else 
  #define STBDS_MAYBE_UNUSED(x) x
#endif

extern void   stbds_hmfree_func(void *p, size_t elemsize, STBDS_MAYBE_UNUSED(size_t keyoff));
```

--- a/stb_ds.h
+++ b/stb_ds.h
@@ -447,6 +447,12 @@ CREDITS
 #define STBDS_FREE(c,p)      free(p)
 #endif
 
+#ifdef _MSC_VER
+#define STBDS_NOTUSED(v)  (void)(v)
+#else
+#define STBDS_NOTUSED(v)  (void)sizeof(v)
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -827,6 +833,7 @@ void stbds_rand_seed(size_t seed)
 
 static size_t stbds_probe_position(size_t hash, size_t slot_count, size_t slot_log2)
 {
+  STBDS_NOTUSED(slot_log2);
   size_t pos;
   pos = hash & (slot_count-1);
   #ifdef STBDS_INTERNAL_BUCKET_START
@@ -1166,6 +1173,7 @@ size_t stbds_hash_bytes(void *p, size_t len, size_t seed)
 
 static int stbds_is_key_equal(void *a, size_t elemsize, void *key, size_t keysize, size_t keyoffset, int mode, size_t i)
 {
+  STBDS_NOTUSED(keyoffset);
   if (mode >= STBDS_HM_STRING)
     return 0==strcmp((char *) key, * (char **) ((char *) a + elemsize*i));
   else

